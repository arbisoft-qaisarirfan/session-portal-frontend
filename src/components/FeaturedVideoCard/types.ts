export type FeaturedVideoCardProps = {
  isVisible?: boolean;
  className?: string;
  event_time: string;
  thumbnail?: string;
  title: string;
  description: string;
  video_duration: number;
  workstream_id: string;
  width?: string;
};
