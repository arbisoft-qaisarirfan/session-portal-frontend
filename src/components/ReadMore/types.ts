export interface ReadMoreProps {
  amountOfWords?: number;
  className?: string;
  showLessText: string;
  showMoreText: string;
  text: string;
}
