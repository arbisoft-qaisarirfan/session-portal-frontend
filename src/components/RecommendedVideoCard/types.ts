export interface RecommendedVideoCardProps {
  className?: string;
  date: string;
  duration: string;
  imgUrl: string;
  onClick?: VoidFunction;
  organizer?: string;
  title: string;
  width?: string;
}
