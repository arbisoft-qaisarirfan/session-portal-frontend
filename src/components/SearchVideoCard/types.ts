export type SearchVideoCardProps = {
  id: number;
  className?: string;
  event_time: string;
  thumbnail?: string;
  title: string;
  description: string;
  video_duration: number;
  workstream_id: string;
  width?: string;
};
