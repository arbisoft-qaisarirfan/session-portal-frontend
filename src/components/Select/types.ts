import { SelectProps as MuiSelectProps, SelectChangeEvent } from "@mui/material/Select";

export type SelectProps = {
  menuItems: { value: string; label: string }[];
  handleChange: (event: SelectChangeEvent) => void;
} & MuiSelectProps;
