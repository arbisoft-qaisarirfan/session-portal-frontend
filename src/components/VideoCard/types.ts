export type VideoCardProps = {
  className?: string;
  data: {
    event_time: string;
    organizer: string;
    thumbnail: string;
    title: string;
    video_duration: string;
  };
  onClick?: VoidFunction;
  width?: string;
};
