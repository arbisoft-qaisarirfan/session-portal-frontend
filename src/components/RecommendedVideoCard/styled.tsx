import Box from "@mui/material/Box";
import Card from "@mui/material/Card";
import { cardContentClasses } from "@mui/material/CardContent";
import { ratingClasses } from "@mui/material/Rating";
import { styled, css, alpha } from "@mui/material/styles";
import { typographyClasses } from "@mui/material/Typography";

import { shouldForwardProp } from "@/utils/styleUtils";

export const RecommendedVideoCardContainer = styled(Card, {
  name: "RecommendedVideoCardContainer",
  shouldForwardProp,
})<{ $width: string }>(({ theme, $width }) => {
  return css`
    background: transparent;
    border-radius: unset;
    cursor: pointer;
    display: inline-flex;
    overflow: unset;
    padding: 0;
    width: ${$width};

    .${cardContentClasses.root} {
      align-items: flex-start;
      display: flex;
      gap: ${theme.spacing(10 / 8)};
      padding: 0;

      :last-child {
        padding-bottom: 10px;
      }

      img {
        border-radius: ${theme.shape.borderRadius + 2}px;
        height: unset;
        width: 113px;
      }

      .video-detail {
        display: flex;
        flex-direction: column;
        gap: 7px;

        .${typographyClasses.h5} {
          -webkit-box-orient: vertical;
          -webkit-line-clamp: 2;
          color: ${theme.palette.colors.white};
          display: -webkit-box;
          font-size: 12px;
          font-style: normal;
          font-weight: 500;
          letter-spacing: 0.4px;
          line-height: normal;
          overflow: hidden;
          text-overflow: ellipsis;
          width: 95%;
        }

        .date-time {
          color: ${theme.palette.colors.gray};
          font-size: 12px;
          font-style: normal;
          font-weight: 500;
          line-height: normal;
          letter-spacing: 0.4px;

          span {
            color: ${theme.palette.colors.white};
          }
        }

        .${ratingClasses.iconFilled} {
          color: ${theme.palette.colors.gold};
        }
      }
    }
  `;
});

export const ImageContainer = styled(Box, {
  name: "ImageContainer",
})(({ theme }) => {
  return css`
    border-radius: ${theme.shape.borderRadius + 2}px;
    height: 73px;
    overflow: hidden;
    position: relative;

    .duration {
      background-color: ${alpha(theme.palette.common.black, 0.7)};
      border-radius: 2px;
      bottom: 6px;
      color: ${theme.palette.colors.white};
      font-size: 12px;
      padding: 2px 4px;
      position: absolute;
      right: 6px;
      z-index: 9999;
    }
  `;
});
