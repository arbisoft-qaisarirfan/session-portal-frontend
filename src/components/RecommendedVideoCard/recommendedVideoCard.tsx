import { FC } from "react";

import Box from "@mui/material/Box";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import Image from "next/image";

import { ImageContainer, RecommendedVideoCardContainer } from "./styled";
import { RecommendedVideoCardProps } from "./types";

const RecommendedVideoCard: FC<RecommendedVideoCardProps> = ({
  className,
  date,
  duration,
  imgUrl,
  title,
  width = "100%",
  onClick,
}) => {
  return (
    <RecommendedVideoCardContainer $width={width} className={className} data-testid="recommended-video-card" onClick={onClick}>
      <CardContent>
        <ImageContainer className="image-container">
          <Image alt={title} height={72} src={imgUrl} width={113} />
          <Typography variant="bodySmall" className="duration" data-testid="duration">
            {duration}
          </Typography>
        </ImageContainer>
        <Box className="video-detail">
          <Typography variant="h5" component="div" title={title}>
            {title}
          </Typography>
          <Typography variant="bodySmall" className="date-time" data-testid="date-time">
            {date}
          </Typography>
        </Box>
      </CardContent>
    </RecommendedVideoCardContainer>
  );
};

export default RecommendedVideoCard;
