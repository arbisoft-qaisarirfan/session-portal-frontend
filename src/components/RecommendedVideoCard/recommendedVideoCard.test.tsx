import { customRender as render, screen } from "@/jest/utils/testUtils";

import RecommendedVideoCard from "./recommendedVideoCard";
import { RecommendedVideoCardProps } from "./types";

const mockProps: RecommendedVideoCardProps = {
  className: "custom-class",
  date: "Jan 01, 2024",
  duration: "10:15",
  imgUrl: "https://example.com/sample.jpg",
  title: "Sample Video Title",
  width: "300px",
};

describe("RecommendedVideoCard", () => {
  it("should renders the component with provided props", () => {
    render(<RecommendedVideoCard {...mockProps} />);

    expect(screen.getByText(mockProps.title)).toBeInTheDocument();
    expect(screen.getByTestId("date-time")).toHaveTextContent(mockProps.date);
    expect(screen.getByTestId("duration")).toHaveTextContent(mockProps.duration);
    const imgUrl = screen.getByRole("img", { name: mockProps.title }).getAttribute("src") ?? "";
    expect(screen.getByRole("img", { name: mockProps.title })).toHaveAttribute("src", imgUrl);
  });

  it("should displays default image when imgUrl is not provided", () => {
    render(<RecommendedVideoCard {...mockProps} imgUrl={undefined} />);
    const imgUrl = screen.getByRole("img", { name: mockProps.title }).getAttribute("src") ?? "";
    expect(screen.getByRole("img", { name: mockProps.title })).toHaveAttribute("src", imgUrl);
  });

  it("should applies custom className if provided", () => {
    render(<RecommendedVideoCard {...mockProps} />);
    expect(screen.getByTestId("recommended-video-card")).toHaveClass(mockProps.className!);
  });

  it("should matches snapshot", () => {
    const { asFragment } = render(<RecommendedVideoCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
