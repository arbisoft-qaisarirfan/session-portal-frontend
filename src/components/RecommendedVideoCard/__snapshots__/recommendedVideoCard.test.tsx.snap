// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendedVideoCard should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root custom-class css-18r2xxk-MuiPaper-root-MuiCard-root-RecommendedVideoCardContainer-root"
    data-testid="recommended-video-card"
    style="--Paper-shadow: var(--mui-shadows-1); --Paper-overlay: var(--mui-overlays-1);"
  >
    <div
      class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
    >
      <img
        alt="Sample Video Title"
        data-nimg="1"
        decoding="async"
        height="72"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fsample.jpg&w=256&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fsample.jpg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fsample.jpg&w=256&q=75 2x"
        style="color: transparent;"
        width="113"
      />
      <div
        class="video-detail MuiBox-root css-0"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-1ayj0vs-MuiTypography-root"
          title="Sample Video Title"
        >
          Sample Video Title
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 date-time css-ljgvyt-MuiTypography-root"
          data-testid="date-time"
        >
          Jan 01, 2024 
          <span>
            / 10:15
          </span>
        </p>
        <span
          aria-label="4 Stars"
          class="MuiRating-root MuiRating-sizeSmall Mui-readOnly MuiRating-readOnly css-1ynetyc-MuiRating-root"
          role="img"
        >
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-8hoymy-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-8hoymy-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-8hoymy-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-8hoymy-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty css-h06ynj-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-8hoymy-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
